### PR TITLE
Add streaming iterator to cursor

### PR DESCRIPTION
**Pull request by zer0n**
See discussion here: #47

### CODE REVIEW

**Comment by zer0n** [comment]
See discussion here: https://github.com/uber/vertica-python/issues/47

**Comment by alexjikim** [comment]
please be aware that this iterator will not _stream_ data. Instead, it will fetch all data into memory and allow for iteration over that set. this would be similar to calling execute() followed by looping over fetchone(). was this your intention?

**Comment by zer0n** [comment]
Oops. So does execute() fetch all data into memory? 

I was under the impression that it didn't and that buffered_rows was a streaming buffer.

**Comment by alexjikim** [comment]
sadly it currently doesnt. execute will either fetch everything into memory (buffered_rows), or use the callback function after each row fetch.

File: vertica_python/vertica/connection.py [hunk 1]
```diff
@@ -57,14 +57,10 @@ def __exit__(self, type, value, traceback):
     #
     # To support vertica_python 0.1.9 interface
     #
-    def query(self, query, handler=None):
```
Comment (zer0n) [inline vertica_python/vertica/connection.py]: With iterate, passing the handler to the ctor is not needed. Having a handler embedded in the cursor also created several inconsistency.

File: vertica_python/vertica/cursor.py [hunk 2]
```diff
@@ -9,15 +9,12 @@
 
 
 class Cursor(object):
-
-    def __init__(self, connection, cursor_type=None, row_handler=None):
```
Comment (zer0n) [inline vertica_python/vertica/cursor.py]: With iterate, passing the handler to the ctor is not needed. Having a handler embedded in the cursor also created several inconsistency.

File: vertica_python/vertica/cursor.py [hunk 3]
```diff
@@ -9,15 +9,12 @@
...
+    def __init__(self, connection, cursor_type=None):
         self.connection = connection
         self.cursor_type = cursor_type
-        self.row_handler = row_handler
         self._closed = False
 
         self.last_execution = None
-        self.buffered_rows = collections.deque()
```
Comment (zer0n) [inline vertica_python/vertica/cursor.py]: buffered_rows is also not needed after the change.

**Pull request merged**